packages/comlink/src/node.ts

```diff
@@ -0,0 +1,430 @@
+              count: 1,
+              responseType: 'disconnect',
+            }),
+          },
+        ],
+        on: {
+          request: {
+            actions: 'create request',
+          },
+          post: {
+            actions: 'buffer message',
+          },
+          disconnect: {
+            target: 'idle',
+          },
+        },
+      },
+      connected: {
+        entry: 'flush buffer',
+        invoke: [
+          {
+            src: 'listen',
+            id: 'listenForMessages',
+            input: listenInputFromContext([MSG_RESPONSE, MSG_HEARTBEAT], {matches: false}),
+          },
+          {
+            src: 'listen',
+            id: 'listenForHeartbeats',
+            input: listenInputFromContext([MSG_HEARTBEAT], {responseType: 'heartbeat.received'}),
+          },
+          {
+            src: 'listen',
+            id: 'listenForDisconnect',
+            input: listenInputFromContext([MSG_DISCONNECT], {
+              count: 1,
+              responseType: 'disconnect',
+            }),
+          },
+        ],
+        on: {
+          'request': {
+            actions: 'create request',
+          },
+          'post': {
+            actions: 'post',
+          },
+          'disconnect': {
+            target: 'idle',
+          },
+          'message.received': {
+            actions: ['send response', 'emit received message'],
+          },
+          'heartbeat.received': {
+            actions: ['send response', 'emit heartbeat'],
+          },
+        },
+      },
+    },
+  })
+  return nodeMachine
+}
+
+/**
+ * @public
+ */
+export const createNode = <R extends Message, S extends Message>(input: NodeInput): Node<R, S> => {
+  const machine = createNodeMachine<R, S>()
+
+  const actor = createActor(machine, {
+    input,
+  })
+
+  const on = <T extends R['type'], U extends Extract<R, {type: T}>>(
+    type: T,
+    handler: (event: U['data']) => U['response'],
+  ) => {
+    const {unsubscribe} = actor.on(
+      // @ts-expect-error @todo `type` typing
+      type,
+      (event: {type: T; message: ProtocolMessage<U>}) => {
+        handler(event.message.data)
+      },
+    )
+    return unsubscribe
+  }
+
+  const onStatus = (handler: (status: string) => void) => {
+    const snapshot = actor.getSnapshot()
+    let currentStatus: string | undefined =
+      typeof snapshot.value === 'string' ? snapshot.value : Object.keys(snapshot.value)[0]
+
+    const {unsubscribe} = actor.subscribe((state) => {
+      const status = typeof state.value === 'string' ? state.value : Object.keys(state.value)[0]
+      if (currentStatus !== status) {
+        currentStatus = status
+        handler(status)
+      }
+    })
+    return unsubscribe
+  }
+
+  const post = (data: WithoutResponse<S>) => {
+    actor.send({type: 'post', data})
+  }
+
+  const fetch = (data: WithoutResponse<S>) => {
+    const resolvable = Promise.withResolvers<S['response']>()
```

Comment:
I'm unsure if our pkg-utils stack is updated to automatically polyfill this yet, I'll make a note to check this before we start shipping comlink to production 😌